glean-core/python/glean/metrics/counter.py

```diff
@@ -51,6 +52,7 @@ def __del__(self):
         if getattr(self, "_handle", 0) != 0:
             _ffi.lib.glean_destroy_counter_metric(self._handle)
 
+    @Dispatcher.launch
```

Comment:
We don't really want the whole function to be dispatched, we only want the [content after the disable check](https://github.com/mozilla/glean/blob/1ee40df622a8084914c8d68210fb4532ffcbea8e/glean-core/android/src/main/java/mozilla/telemetry/glean/private/CounterMetricType.kt#L74) to be dispatched. While this doesn't matter much _right now_, this might create inconsistencies in the behaviour when we add asynchrony or new metric types.

Reply:
Good point -- had to refactor how the dispatchers work to make this clean, but it should work.